src/Spice86.Core/Emulator/OperatingSystem/LocalFileSearchManager.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace Spice86.Core.Emulator.OperatingSystem;
+
+using System.Linq;
+
+public class LocalFileSearchManager {
+    public string[] FindFilesUsingWildCmp(string searchFolder, string searchPattern,
+        EnumerationOptions enumerationOptions) {
+        return Directory.EnumerateFileSystemEntries(searchFolder, "*", enumerationOptions)
+            .Where(path => WildFileCmp(Path.GetFileName(path), searchPattern))
+            .ToArray();
+    }
+
+    public static bool WildFileCmp(ReadOnlySpan<char> fileSpan, ReadOnlySpan<char> wildSpan) {
+        int fDotPos = fileSpan.IndexOf('.');
+        ReadOnlySpan<char> fileName = fDotPos >= 0 ? fileSpan[..fDotPos] : fileSpan;
+        ReadOnlySpan<char> fileExt = fDotPos >= 0 ? fileSpan[(fDotPos + 1)..] : ReadOnlySpan<char>.Empty;
+
+        int wDotPos = wildSpan.IndexOf('.');
+        ReadOnlySpan<char> wildName = wDotPos >= 0 ? wildSpan[..wDotPos] : wildSpan;
+        ReadOnlySpan<char> wildExt = wDotPos >= 0 ? wildSpan[(wDotPos + 1)..] : ReadOnlySpan<char>.Empty;
+
+        if (wDotPos < 0) {
+            wildExt = "???".AsSpan();
+        }
+        
+        return Matches83Part(fileName, wildName, 8) && Matches83Part(fileExt, wildExt, 3);
+    }
+
+    private static bool Matches83Part(ReadOnlySpan<char> filePart, ReadOnlySpan<char> wildPart, int maxLen) {
+        bool sawStar = false;
+        for (int i = 0; i < maxLen; i++) {
+            char wc = i < wildPart.Length ? wildPart[i] : ' ';
```

Comment:
Are you sure the matching has to be done with spaces if wildcard is less than filename length? Maybe document that in a code comment?
I would create a function to extract char i or space to simplify common logic on line 32 and 39

Reply:
I re-implemented the version of DosBox Staging and extended the test cases.
Please have a look again.